### PR TITLE
Fix Issue #3139 (parfors reuse of reduction variable across prange blocks)

### PR DESCRIPTION
**Pull request by DrTodd13**
Fix for #3139

### CODE REVIEW

File: numba/npyufunc/parfor.py [hunk 1]
```diff
@@ -371,7 +371,10 @@ def _create_gufunc_for_parfor_body(
     for var in parfor_redvars:
         arr = var + "_arr"
         parfor_redarrs.append(arr)
-        typemap[arr] = types.npytypes.Array(typemap[var], 1, "C")
+        if arr in typemap:
+            assert(typemap[arr] == types.npytypes.Array(typemap[var], 1, "C"))
```
Comment (stuartarchibald) [inline numba/npyufunc/parfor.py]: Will this work once the array reduction PR goes in?

Comment (DrTodd13) [reply numba/npyufunc/parfor.py]: No.  So, we should wait until that other PR merges then rebase here.  Once that happens, the fix is easy.

Comment (stuartarchibald) [reply numba/npyufunc/parfor.py]: Ok, great, thanks. Let's wait on that.

**Comment by stuartarchibald** [comment]
Currently waiting on https://github.com/numba/numba/pull/3127

File: numba/tests/test_parfors.py [hunk 2]
```diff
@@ -1174,6 +1174,20 @@ def test_impl(n):
         # as a different name
         self.assertEqual(countArrayAllocs(test_impl, (types.intp,)), 1)
 
+    @skip_unsupported
+    def test_reduction_var_reuse(self):
```
Comment (stuartarchibald) [inline numba/tests/test_parfors.py]: Once array reduction support is merge I think a test similar to this but with `acc` as an array would be beneficial. Thanks.

Comment (DrTodd13) [reply numba/tests/test_parfors.py]: Done.

File: numba/npyufunc/parfor.py (hunk 1 above)
Comment (DrTodd13) [reply numba/npyufunc/parfor.py]: Done.

File: numba/tests/test_parfors.py (hunk 2 above)
Comment (DrTodd13) [reply numba/tests/test_parfors.py]: Done.

Comment (stuartarchibald) [reply numba/tests/test_parfors.py]: Thanks.

**Comment by stuartarchibald** [comment]
Thanks for the patch, this looks good, will just run it through the build farm. The test fail on Travis CI is entirely unrelated to the change here.

**Comment by stuartarchibald** [comment]
Build farm passed. This can be merged as convenient. Thanks again.

**Pull request merged**